File: M_life/src/main/java/com/m_life/m_life/controller/PostLikeController.java
package com.m_life.m_life.controller;

import com.m_life.m_life.domain.UserAccount;
import com.m_life.m_life.service.CustomUserDetails;
import com.m_life.m_life.service.PostLikeService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/post/{postId}/like")
@RequiredArgsConstructor
public class PostLikeController {
    private final PostLikeService postLikeService;


    @PostMapping
    public ResponseEntity<Void> likePost(@PathVariable(name = "postId") Long postId,
                                         @AuthenticationPrincipal CustomUserDetails userDetails) {

        UserAccount userAccount = userDetails.getUserAccount();
        postLikeService.likePost(postId, userAccount);
        return ResponseEntity.ok().build();
    }

    @DeleteMapping
    public ResponseEntity<Void> unlikePost(@PathVariable(name = "postId") Long postId,
                                           @AuthenticationPrincipal CustomUserDetails userDetails) {
        UserAccount userAccount = userDetails.getUserAccount();
        postLikeService.unlikePost(postId, userAccount);
        return ResponseEntity.ok().build();
    }


    @GetMapping("/liked")
    public ResponseEntity<Boolean> isLikedByCurrentUser(@PathVariable(name = "postId") Long postId,
                                                        @AuthenticationPrincipal CustomUserDetails userDetails) {
        UserAccount userAccount = userDetails.getUserAccount();
        boolean isLiked = postLikeService.isLikedByUser(postId, userAccount);
        return ResponseEntity.ok(isLiked);
    }
}

File: flutterworkspace/m_life_app/lib/controller/postLike_controller.dart
import 'package:get/get.dart';
import 'package:m_life_app/domain/postLike/postLike_repository.dart';

class PostLikeController extends GetxController {
  final PostLikeRepository _postLikeRepository = Get.put(PostLikeRepository());

  final RxBool isLiked = false.obs; // UI가 관찰 가능한 변수 => 변경이 되며 UI가 자동으로 업데이트
  final int postId;

  PostLikeController(this.postId);

  @override
  void onInit() {
    super.onInit();
    isLikedByCurrentUser(postId);
  }

  Future<void> isLikedByCurrentUser(int postId) async {
    bool isLiked = await _postLikeRepository.isLikedByCurrentUser(postId);
    this.isLiked.value = isLiked;
  }

  Future<void> likePost(int postId) async {
    bool isLiked = await _postLikeRepository.likePost(postId);
    if (isLiked == true) {
      this.isLiked.value = true;
    }
  }

  Future<void> unlikePost(int postId) async {
    bool isLiked = await _postLikeRepository.unlikePost(postId);
    if (isLiked == true) {
      this.isLiked.value = false;
    }
  }
}
